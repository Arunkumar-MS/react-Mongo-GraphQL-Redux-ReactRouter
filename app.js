import express from 'express';
import graphQLHTTP from 'express-graphql';
import Schema from './schema';

var app = express()
app.set('views', __dirname);
app.engine('html', require('ejs').renderFile);
app.set('view engine', 'ejs');
app.use(express.static(__dirname));
app.listen((process.env.PORT || 4000), (err) => {
    if (err) {
        console.error(err)
        return
    }
    console.log(`Server is now running on localhost: ${process.env.PORT || 4000}`)
})

app.get('/', function(req, res) {
    res.render('index.html');
});
app.use('/api', graphQLHTTP({
    schema: Schema,
    pretty: true,
    graphiql: true
}));
