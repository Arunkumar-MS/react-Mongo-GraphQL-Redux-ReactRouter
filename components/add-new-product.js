import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import {addNewProduct} from '../actions/product';
import {RESET_FlAGS} from '../constents';

class AddProduct extends Component {
  addProduct = () => {
    const args = {
      title:this.refs.title.value,
      url: this.refs.url.value,
      discription:this.refs.discription.value,
      price: this.refs.price.value,
      quantity: this.refs.qty.value
    };
    addNewProduct(args, this.props.dispatch);

  }
  resetToDefault = () => {
    this.refs.title.value = null;
    this.refs.url.value = null;
    this.refs.discription.value= null;
    this.refs.price.value= null;
    this.refs.qty.value= null;

  }
  componentDidMount() {
    this.props.dispatch({type: RESET_FlAGS});
  }
  componentWillReceiveProps(nextProps) {
    nextProps.added ? this.resetToDefault() : null;
  }
  render() {

    return(
      <div className="main-add-product-wrapper">
        <div className="main-add-product">
        <table>
        <tbody>
          <tr>
            <td>  <label> Product Title </label></td>
            <td><input ref ="title" type="text" className="form-control" /></td>
          </tr>
          <tr>
            <td>  <label> Product Price </label></td>
            <td><input ref ="price" type="text" className="form-control" /></td>
          </tr>
          <tr>
            <td>  <label> Product Quantity </label></td>
            <td><input ref ="qty" type="text" className="form-control" /></td>
          </tr>
          <tr>
            <td>  <label> Product Url </label></td>
            <td><input ref ="url" type="text" className="form-control" /></td>
          </tr>
          <tr>
            <td>  <label> Product discription </label></td>
            <td><textarea ref ="discription" rows="4" className="form-control" cols="50"></textarea>
            </td>
          </tr>
        </tbody>
        </table>
        <div className="submit-ctl">
          <button onClick = {() => this.addProduct()}type="button" className="btn btn-primary addBtn">Add</button>
          <Link className="btn btn-primary" to="/"> Back to Home </Link>
        </div>
        {this.props.added ? (<label> Added Succefully </label>) : null}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  added: state.product.added
});

const mapDispatchToProps = dispatch => ({
    dispatch
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddProduct)
