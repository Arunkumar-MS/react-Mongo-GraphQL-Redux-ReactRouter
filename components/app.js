import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { getProduct } from '../actions/product';
import List from './list';

export class App extends Component {

  componentDidMount() {
    getProduct(this.props.dispatch);
  }
  componentWillReceiveProps(nextProps) {
    document.body.classList.toggle('overlay', nextProps.isLoading)
  }
  componentWillUnmount() {
  document.body.classList.remove('overlay')
  }
  render() {
    const cssClassName = this.props.isLoading ? 'loader' : null;
    return(
        <div className="main-table-list">
          <Link to="/add" className="add-new btn btn-primary">Add new Product ></Link>
          <List products ={this.props.products}/>
          <div className={cssClassName}></div>
        </div>
    );
  }
}

const mapStateToProps = state => ({
  products: state.product.products,
  isLoading: state.product.isLoading
});

const mapDispatchToProps = dispatch => ({
    dispatch
});

App.propTypes = {
  products: PropTypes.array.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
