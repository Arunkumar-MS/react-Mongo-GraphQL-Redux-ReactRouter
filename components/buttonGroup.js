import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import {
    DisableDelete
} from '../config';
import {
    deleteProduct,
    getProductById
} from '../actions/product';
 class ButtonActions extends Component {
  static propTypes ={
    row : PropTypes.object
  };
  deleteItem = (id) =>{
    deleteProduct({id}, this.props.dispatch);
  };
  viewDetails = (id) =>{
    getProductById({id}, this.props.dispatch);
  }
  render(){
    const id = this.props.row.id;
    return(
      <div className="btn-group">
      <Link to={`/edit/${id}`} className="btn btn-primary">Edit</Link>
      <Link to={`/product-overview/${id}`} className="btn btn-primary">View</Link>
      <button onClick={() => this.deleteItem(id)} disabled={ DisableDelete } type="button" className="btn btn-primary">Delete</button>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
    dispatch
});


export default connect(
  mapDispatchToProps
)(ButtonActions)
