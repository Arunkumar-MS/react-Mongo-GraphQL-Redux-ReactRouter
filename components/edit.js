import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import {
    getProductById,
    updateProduct
} from '../actions/product';

import {
    RESET_FlAGS
} from '../constents';

class Edit extends Component {
  constructor(props){
    super();
    this.state = {
      product: props.product
    };
  }
  editProduct = () =>{
    console.log(this.state.product);
    updateProduct(this.state.product, this.props.dispatch);
  }
  dataChanged = (data) => {
         this.setState({ product : {...this.state.product, ...data}})
     }
  componentWillReceiveProps(nextProps) {
    this.setState({
      product: nextProps.product
    });
  }
  componentDidMount() {
    const args = {
      id: this.props.params.id
    };
    getProductById(args, this.props.dispatch);
    this.props.dispatch({type: RESET_FlAGS});
  }
  render() {
    const product = this.state.product;
    return(
      <div className="main-add-product-wrapper">
        <div className="main-add-product">
        <table>
        <tbody>
          <tr>
            <td>  <label> Product Title </label></td>
            <td><input onChange={ (event) => this.dataChanged({title: event.target.value})} type="text" className="form-control" value={this.state.product.title} /></td>
          </tr>
          <tr>
            <td>  <label> Product Price </label></td>
            <td><input onChange={ (event) => this.dataChanged({price: event.target.value})} type="text" className="form-control" value={this.state.product.price} /></td>
          </tr>
          <tr>
            <td>  <label> Product Quantity </label></td>
            <td><input onChange={ (event) => this.dataChanged({quantity: event.target.value})} type="text" className="form-control" value={this.state.product.quantity}  /></td>
          </tr>
          <tr>
            <td>  <label> Product Url </label></td>
            <td><input onChange={ (event) => this.dataChanged({url: event.target.value})} type="text" className="form-control" value={this.state.product.url} /></td>
          </tr>
          <tr>
            <td>  <label> Product discription </label></td>
            <td><textarea onChange={ (event) => this.dataChanged({discription: event.target.value})} rows="4" className="form-control" cols="50" value={this.state.product.discription}  ></textarea>
            </td>
          </tr>
        </tbody>
        </table>
        <div className="submit-ctl">
          <button onClick = {() => this.editProduct()}type="button" className="btn btn-primary addBtn">Update</button>
          <Link className="btn btn-primary" to="/"> Back to Home </Link>
        </div>
        {this.props.added ? (<label> Updated Succefully </label>) : null}
        </div>

      </div>
    );
  }
}

const mapStateToProps = state => ({
  added: state.product.added,
  product: state.product.product
});

const mapDispatchToProps = dispatch => ({
    dispatch
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Edit)
