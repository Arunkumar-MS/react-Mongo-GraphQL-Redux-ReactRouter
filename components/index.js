import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import {Router, Route, browserHistory } from 'react-router';
import reducers from '../reducers';
import App from './app';
import AddProduct from './add-new-product';
import ProductOverview from './product-overview';
import Edit from './edit';

const store = createStore(reducers);

render(
  <Provider store={store}>
    <Router history={browserHistory}>
      <Route path="/" component={App} />
      <Route path="/add" component={AddProduct} />
      <Route path="/edit/:id" component={Edit} />
      <Route path="/product-overview/:id" component={ProductOverview} />
    </Router>
  </Provider>,
  document.getElementById('app')
);
