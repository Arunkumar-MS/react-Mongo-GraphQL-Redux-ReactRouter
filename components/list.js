import React, { Component, PropTypes } from 'react';
import TableRow from './table-row';
export default class List extends Component {
  static propTypes ={
    products : PropTypes.array
  }
  render(){

    return(
      <table className = "table table-striped product-list">
      <thead>
         <tr>
            <th>Title</th>
            <th>Price</th>
            <th>Quantity</th>
            <th>Actions</th>
         </tr>
      </thead>
      <tbody>
          {!!this.props.products ? this.props.products.map((item) => {
              return (<TableRow key={item.id} row={item}/>);
          }): null}
      </tbody>
      </table>
    );
  }
}
