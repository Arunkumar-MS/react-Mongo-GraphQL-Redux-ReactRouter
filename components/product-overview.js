import React, { Component, PropTypes } from 'react';
import { Link , browserHistory} from 'react-router';
import { connect } from 'react-redux';
import {
    DisableDelete
} from '../config';
import {
    deleteProduct,
    getProductById
} from '../actions/product';

class ProductOverview extends Component {
  componentDidMount() {
    const args = {
      id: this.props.params.id
    };
    getProductById(args, this.props.dispatch);

  }
  componentWillReceiveProps(nextProps) {
    document.body.classList.toggle('overlay', nextProps.isLoading)
  }
  componentWillUnmount() {
  document.body.classList.remove('overlay')
  }
  deleteItem = (id) => {
    deleteProduct({id}, this.props.dispatch);
    browserHistory.goBack();
  };
  render() {
    const cssClassName = this.props.isLoading ? 'loader' : null;
    const product = this.props.product;
    return(
      <div className="main-product-overview-wrapper">
      <div className="product-overview">
        <div className="product-image">
        <img src={product.url}/>
        </div>
        <div>
          <h3>{product.title}</h3>
          <div>  Price :{product.price} </div>
          <div>  quantity :{product.quantity} </div>
        </div>
        </div>
        <div className="discription">
          <strong>Product discription</strong>
          <p>{product.discription}</p>

          <div className="overview-submit-ctl">
            <Link to={`/edit/${this.props.product.id}`} className="btn btn-primary">Edit</Link>
            <button onClick={() => this.deleteItem(this.props.params.id)} disabled={ DisableDelete } type="button" className="btn btn-primary">Delete</button>
            <Link className="btn btn-primary" to="/"> Back to Home </Link>
          </div>
        </div>
        <div className={cssClassName}></div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  product: state.product.product,
  isLoading: state.product.isLoading
});

const mapDispatchToProps = dispatch => ({
    dispatch
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductOverview)
