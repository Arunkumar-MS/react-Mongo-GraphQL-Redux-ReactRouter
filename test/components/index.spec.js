import { App } from '../../components/app';

describe('<App />', () => {
  let props;
  beforeEach(() => {
    props = {
      products: 
       [
        {
          "id": "5945ca3cfb7fe7000460e0da",
          "title": "New opne",
          "quantity": 3,
          "price": 12
        },
        {
          "id": "589a01417fbb7f00111d0892",
          "title": "Broccoli",
          "quantity": 5000,
          "price": 100
        },
        {
          "id": "58bef54b4a4b4e0004112d7c",
          "title": "Wine",
          "quantity": 20,
          "price": 1000
        },
        {
          "id": "58bef5c44a4b4e0004112d7d",
          "title": "Military backpack",
          "quantity": 200,
          "price": 2000
        },
        {
          "id": "5941942400dc660004acd152",
          "title": "Product Title",
          "quantity": 123123,
          "price": 121
        }
      ]
    }
  });
  test('should render App Component', () => {
    const component = shallow(<App { ...props }/>);
    expect(component).toMatchSnapshot();
  });
});
