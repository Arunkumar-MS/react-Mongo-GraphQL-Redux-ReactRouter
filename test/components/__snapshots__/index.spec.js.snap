// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="main-table-list"
>
    <Link
        className="add-new btn btn-primary"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/add"
    >
        Add new Product &gt;
    </Link>
    <List
        products={
            Array [
                Object {
                  "id": "5945ca3cfb7fe7000460e0da",
                  "price": 12,
                  "quantity": 3,
                  "title": "New opne",
                },
                Object {
                  "id": "589a01417fbb7f00111d0892",
                  "price": 100,
                  "quantity": 5000,
                  "title": "Broccoli",
                },
                Object {
                  "id": "58bef54b4a4b4e0004112d7c",
                  "price": 1000,
                  "quantity": 20,
                  "title": "Wine",
                },
                Object {
                  "id": "58bef5c44a4b4e0004112d7d",
                  "price": 2000,
                  "quantity": 200,
                  "title": "Military backpack",
                },
                Object {
                  "id": "5941942400dc660004acd152",
                  "price": 121,
                  "quantity": 123123,
                  "title": "Product Title",
                },
              ]
        }
    />
    <div
        className={null}
    />
</div>,
  "nodes": Array [
    <div
      className="main-table-list"
>
      <Link
            className="add-new btn btn-primary"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/add"
      >
            Add new Product &gt;
      </Link>
      <List
            products={
                  Array [
                        Object {
                          "id": "5945ca3cfb7fe7000460e0da",
                          "price": 12,
                          "quantity": 3,
                          "title": "New opne",
                        },
                        Object {
                          "id": "589a01417fbb7f00111d0892",
                          "price": 100,
                          "quantity": 5000,
                          "title": "Broccoli",
                        },
                        Object {
                          "id": "58bef54b4a4b4e0004112d7c",
                          "price": 1000,
                          "quantity": 20,
                          "title": "Wine",
                        },
                        Object {
                          "id": "58bef5c44a4b4e0004112d7d",
                          "price": 2000,
                          "quantity": 200,
                          "title": "Military backpack",
                        },
                        Object {
                          "id": "5941942400dc660004acd152",
                          "price": 121,
                          "quantity": 123123,
                          "title": "Product Title",
                        },
                      ]
            }
      />
      <div
            className={null}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App
        products={
                Array [
                        Object {
                          "id": "5945ca3cfb7fe7000460e0da",
                          "price": 12,
                          "quantity": 3,
                          "title": "New opne",
                        },
                        Object {
                          "id": "589a01417fbb7f00111d0892",
                          "price": 100,
                          "quantity": 5000,
                          "title": "Broccoli",
                        },
                        Object {
                          "id": "58bef54b4a4b4e0004112d7c",
                          "price": 1000,
                          "quantity": 20,
                          "title": "Wine",
                        },
                        Object {
                          "id": "58bef5c44a4b4e0004112d7d",
                          "price": 2000,
                          "quantity": 200,
                          "title": "Military backpack",
                        },
                        Object {
                          "id": "5941942400dc660004acd152",
                          "price": 121,
                          "quantity": 123123,
                          "title": "Product Title",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "products": Array [
            Object {
              "id": "5945ca3cfb7fe7000460e0da",
              "price": 12,
              "quantity": 3,
              "title": "New opne",
            },
            Object {
              "id": "589a01417fbb7f00111d0892",
              "price": 100,
              "quantity": 5000,
              "title": "Broccoli",
            },
            Object {
              "id": "58bef54b4a4b4e0004112d7c",
              "price": 1000,
              "quantity": 20,
              "title": "Wine",
            },
            Object {
              "id": "58bef5c44a4b4e0004112d7d",
              "price": 2000,
              "quantity": 200,
              "title": "Military backpack",
            },
            Object {
              "id": "5941942400dc660004acd152",
              "price": 121,
              "quantity": 123123,
              "title": "Product Title",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="main-table-list"
>
          <Link
                    className="add-new btn btn-primary"
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/add"
          >
                    Add new Product &gt;
          </Link>
          <List
                    products={
                              Array [
                                        Object {
                                          "id": "5945ca3cfb7fe7000460e0da",
                                          "price": 12,
                                          "quantity": 3,
                                          "title": "New opne",
                                        },
                                        Object {
                                          "id": "589a01417fbb7f00111d0892",
                                          "price": 100,
                                          "quantity": 5000,
                                          "title": "Broccoli",
                                        },
                                        Object {
                                          "id": "58bef54b4a4b4e0004112d7c",
                                          "price": 1000,
                                          "quantity": 20,
                                          "title": "Wine",
                                        },
                                        Object {
                                          "id": "58bef5c44a4b4e0004112d7d",
                                          "price": 2000,
                                          "quantity": 200,
                                          "title": "Military backpack",
                                        },
                                        Object {
                                          "id": "5941942400dc660004acd152",
                                          "price": 121,
                                          "quantity": 123123,
                                          "title": "Product Title",
                                        },
                                      ]
                    }
          />
          <div
                    className={null}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="main-table-list"
>
          <Link
                    className="add-new btn btn-primary"
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/add"
          >
                    Add new Product &gt;
          </Link>
          <List
                    products={
                              Array [
                                        Object {
                                          "id": "5945ca3cfb7fe7000460e0da",
                                          "price": 12,
                                          "quantity": 3,
                                          "title": "New opne",
                                        },
                                        Object {
                                          "id": "589a01417fbb7f00111d0892",
                                          "price": 100,
                                          "quantity": 5000,
                                          "title": "Broccoli",
                                        },
                                        Object {
                                          "id": "58bef54b4a4b4e0004112d7c",
                                          "price": 1000,
                                          "quantity": 20,
                                          "title": "Wine",
                                        },
                                        Object {
                                          "id": "58bef5c44a4b4e0004112d7d",
                                          "price": 2000,
                                          "quantity": 200,
                                          "title": "Military backpack",
                                        },
                                        Object {
                                          "id": "5941942400dc660004acd152",
                                          "price": 121,
                                          "quantity": 123123,
                                          "title": "Product Title",
                                        },
                                      ]
                    }
          />
          <div
                    className={null}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App
    products={
        Array [
            Object {
              "id": "5945ca3cfb7fe7000460e0da",
              "price": 12,
              "quantity": 3,
              "title": "New opne",
            },
            Object {
              "id": "589a01417fbb7f00111d0892",
              "price": 100,
              "quantity": 5000,
              "title": "Broccoli",
            },
            Object {
              "id": "58bef54b4a4b4e0004112d7c",
              "price": 1000,
              "quantity": 20,
              "title": "Wine",
            },
            Object {
              "id": "58bef5c44a4b4e0004112d7d",
              "price": 2000,
              "quantity": 200,
              "title": "Military backpack",
            },
            Object {
              "id": "5941942400dc660004acd152",
              "price": 121,
              "quantity": 123123,
              "title": "Product Title",
            },
          ]
    }
/>,
}
`;
